main.go

```diff
@@ -259,6 +260,14 @@ func main() {
 		builder.SetCustomOptions(options...)
 	}
 
+	if configs.BuildSystem == "legacy" {
+		legacyQuery := "--buildFlag='-UseModernBuildSystem=0'"
+		builder.SetCustomOptions(legacyQuery)
+	} else if configs.BuildSystem == "modern" {
+		modernQuery := "--buildFlag='-UseModernBuildSystem=YES'"
```

Comment:
For consistency, please use `YES` and `NO` or `0` and `1`, but let's not mix the two. According to this SO answer, both formats work: https://stackoverflow.com/a/51205222